app/models/post_viewer.rb

```diff
@@ -3,4 +3,5 @@ class PostViewer < ActiveRecord::Base
   belongs_to :user
 
   validates_presence_of :user, :post
+  attr_accessible :user, :user_id, :post, :post_id
```

Comment:
Otherwise it fails to add `post_id` on creation through setting `viewer_ids` – not sure if there's a simpler way around this.